.github/workflows/wsl-exec.yaml

```diff
@@ -0,0 +1,184 @@
+name: Execute bash scripts in WSL
+
+concurrency: azure-vm
+
+on:
+  workflow_call:
+    secrets:
+      az_creds:
+        description: Credentials to use for Azure
+        required: true
+      az_vm_hostname:
+        description: Hostname for "xfreerdp /v:az_vm_hostname"
+        required: true
+      az_vm_username:
+        description: Username for "xfreerdp /u:az_vm_username"
+        required: true
+      az_vm_password:
+        description: Password for "xfreerdp /p:az_vm_password"
+        required: true
+      certificate:
+        description: Certificate used to enable connecting to Azure
+        required: true
+      certificate_password:
+        description: Pasword to the secret certificate
+        required: true
+    inputs:
+      exec:
+        description: Script to run on WSL
+        required: true
+        type: string
+      repo:
+        description: The url to clone the repo from
+        required: true
```

Comment:
Do we necessarily needs a repo for executing a script (as it can be an inline exec script)?

Reply:
Fair enough. I made it optional so that cloning the repo can be skipped.

It's still useful because most CI runs will need to clone the repo in order to run tests on it.

Reply:
Reverted this for simplicity. Repository is deduced automatically and as such there is no need for these inputs.